test/BUILD

```diff
@@ -20,6 +20,7 @@ envoy_cc_test_library(
     deps = [
         "//source/common/common:thread_lib",
         "//source/common/event:libevent_lib",
+        "//source/exe:sigaction_lib",
```

Comment:
Should we select here since we do the ifdef in the impl file? (I'm also fine just dropping the ifdef in test main. Not sure why we wouldn't always have it).